File: backend/src/main/java/Money/Fact/FactDto.java
package Money.Fact;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FactDto {
    private Long id;
    private String name;              // наименование
    private String factNumber;
    private LocalDate date;           // дата
    private BigDecimal cost;          // стоимость
    private BigDecimal amount;        // количество
    private Long unitId;              // id единицы измерения
    private Long objectId;            // id объекта
    private Long contractId;          // id договора
    private Long pnlId;               // id статьи (PnL)
    private String basis;             // основание
    private String description;       // описание
}

File: frontend/src/fact/DTO/FactDTO.ts


export interface FactDTO{
  id: number;
  accepted: boolean;
  name: string;
  factNumber: string;
  date: string;
  cost: number;
  actualVAT?: number;
  amount?: number;
  unit?: {
    id: number;
    name: string;
  };
  object?: {
    id: number;
    name: string;
  };
  contract?: {
    id: number;
    name: string;
    actualCostWithoutVAT?: number;
    actualVAT?: number;
    contractor?: string ;
  };
  pnl?: {
    id: number;
    name: string;
  };
  basis?: string;
  description?: string;

}
